data-prepper-script/src/test/java/org/opensearch/dataprepper/script/parser/ScriptParserTest.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.script.parser;
+
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.tree.ParseTreeWalker;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.opensearch.dataprepper.script.parser.util.ListenerMatcher;
+import org.opensearch.dataprepper.script.parser.util.TestListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+class ScriptParserTest {
+    private static final Logger LOG = LoggerFactory.getLogger(ScriptParserTest.class);
+
+    private ScriptParser scriptParser;
+    private TestListener listener;
+    private ParseTreeWalker walker;
+
+    @BeforeEach
+    public void beforeEach() {
+        scriptParser = new ScriptParser();
+        listener = new TestListener();
+        walker = new ParseTreeWalker();
+    }
+
+    private void parseStatement(final String statement) {
+        final ParseTree parseTree = scriptParser.parse(statement);
+        walker.walk(listener, parseTree);
+
+        LOG.info("JSON:\n{}", listener.toPrettyString());
```

Comment:
I am concerned with this testings approach. TestListener has no unit tests at the moment and we are relying on it to validate code. A bug could be introduced in ScriptParser and hidden by TestListener. 

Would it be easier to just validate the parseTree object itself over providing test translating layer that we should validate as well?